lib/ecto/changeset.ex

```diff
@@ -1822,7 +1822,8 @@ defmodule Ecto.Changeset do
 
   """
   @spec unsafe_validate_unique(t, atom | [atom, ...], Ecto.Repo.t, Keyword.t) :: t
-  def unsafe_validate_unique(changeset, fields, repo, opts \\ []) when is_list(opts) do
+  def unsafe_validate_unique(changeset, fields, repo, opts \\ [], repo_opts \\ [])
```

Comment:
Instead of passing it as a separate argument, pass it as part of `opts`. So here you do:

    repo_opts = Keyword.pop(opts, :repo_opts, [])

:)